protostar/commands/test/test_result_structured_formatter.py

```diff
@@ -0,0 +1,193 @@
+        result["fuzz_runs"] = passed_fuzz_test_case_result.fuzz_runs_count
+
+    if passed_fuzz_test_case_result.execution_resources:
+        if passed_fuzz_test_case_result.execution_resources.estimated_gas is not None:
+            result[
+                "gas"
+            ] = passed_fuzz_test_case_result.execution_resources.estimated_gas
+        if passed_fuzz_test_case_result.execution_resources.n_steps:
+            result["steps"] = str(
+                passed_fuzz_test_case_result.execution_resources.n_steps
+            )
+        if passed_fuzz_test_case_result.execution_resources.n_memory_holes:
+            result[
+                "memory_holes"
+            ] = passed_fuzz_test_case_result.execution_resources.n_memory_holes
+
+    if passed_fuzz_test_case_result.execution_resources:
+        result["builtins"] = [
+            {"name": name, "count": count}
+            for name, count in passed_fuzz_test_case_result.execution_resources.builtin_name_to_count_map.items()
+        ]
+
+    return json.dumps(result)
+
+
+def _format_skipped_test_case_result(skipped_test_case_result: SkippedTestCaseResult):
+    result: PythonData = {
+        "type": "test_case_result",
+        "status": "skipped",
+        "test_suite_path": str(skipped_test_case_result.file_path),
+        "test_case_name": skipped_test_case_result.test_case_name,
+    }
+
+    reason = skipped_test_case_result.reason
+    if reason is not None:
+        result["reason"] = reason
+
+    return json.dumps(result)
+
+
+def _format_failed_fuzz_test_case_result(
+    failed_fuzz_test_case_result: FailedFuzzTestCaseResult,
+) -> str:
+    return _format_failed_test_case_result(failed_fuzz_test_case_result)
+
+
+def _format_broken_fuzz_test_case_result(
+    broken_fuzz_test_case_result: BrokenFuzzTestCaseResult,
+) -> str:
+    return _format_broken_test_case_result(broken_fuzz_test_case_result)
+
+
+def _format_broken_test_suite_result(
+    broken_test_suite_result: BrokenTestSuiteResult,
+) -> str:
+    result: PythonData = {
+        "type": "test_case_result",
+        "status": "broken",
+        "test_suite_path": str(broken_test_suite_result.file_path),
+        "exception": broken_test_suite_result.exception,
+    }
+
+    return json.dumps(result)
+
+
+def _format_unexpected_exception_test_suite_result(
+    unexpected_exception_test_suite_result: UnexpectedBrokenTestSuiteResult,
+) -> str:
+    result: PythonData = {
+        "type": "test_case_result",
+        "status": "unexpected_exception",
+        "test_suite_path": str(unexpected_exception_test_suite_result.file_path),
+        "traceback": unexpected_exception_test_suite_result.traceback,
+        "exception": str(unexpected_exception_test_suite_result.exception),
+        "protostar_message": UNEXPECTED_PROTOSTAR_ERROR_MSG,
+    }
+
+    return json.dumps(result)
+
+
+def get_formatted_execution_time(execution_time: float) -> float:
```

Comment:
What is the purpose of this function? If rounding to 2dec places then you can use 
`"{:.2f}".format(execution_time)` instead.

Reply:
I know, I first did this with the `format` function but back then I wanted to have floats there, now we can use string right away I think.

Reply:
[format + names + remove redundant function](https://github.com/software-mansion/protostar/pull/1254/commits/f62b551533db9762d7ce549c5fa2823cc8965a14)